bitnami/external-dns/templates/deployment.yaml

```
@@ -346,12 +346,18 @@ spec:
             - --txt-encrypt-enabled
             - --txt-encrypt-aes-key=$(TXT_AES_ENCRYPTION_KEY)
             {{- end }}
-            {{- range $key, $value := .Values.extraArgs }}
-             # Extra arguments
-              {{- if $value }}
+            {{- if .Values.extraArgs }}
+              {{- range $key, $value := .Values.extraArgs }}
+            # Extra arguments
+                {{- if kindIs "slice" $value }}
+                  {{- range $val := $value }}
+            - --{{ $key }}={{ $val }}
+                  {{- end }}
+                {{- else if $value }}
             - --{{ $key }}={{ $value }}
-              {{- else }}
+                {{- else }}
             - --{{ $key }}
+                {{- end }}
```

Not sure why we are enforcing users to use a map or yaml structure here. I think we should keep it as simple and flexible as possible to avoid these problems. Something like this would cover user needs and it doesn't need to rewrite external-dns args in yaml format:
```suggestion
            {{- if .Values.extraArgs }}
            {{ ( include "common.tplvalues.render" .Values.extraArgs . )| indent 12  | trim }}
            {{- end }}
```
I'd create a new major version changing the type of the `extraArgs` value and explaining the change in the README file.

---

Although I haven't experimented with the method you mentioned, it does simplify the extraArgs